### ISSUE TITLE
#[allow(clippy::shadow_unrelated)]

### ISSUE THREAD

**Issue by whatisaphone** [rust-lang/rust-clippy#5455]
It looks like `shadow_unrelated` cannot be suppressed by annotating the variable:

```rust
#![warn(clippy::shadow_unrelated)]

fn abc() {
    let x = 1;
    #[allow(clippy::shadow_unrelated)]
    let x = 2;
}
```

**Expected result:**

No clippy warnings, since the lint is `allow`ed.

**Actual result:**

<details>
<summary>Clippy emits a warning.</summary>

```text
warning: `x` is shadowed by `2`
 --> src\agents\tata\mod.rs:6:13
  |
6 |         let x = 2;
  |             ^
  |
note: lint level defined here
 --> src\agents\tata\mod.rs:1:9
  |
1 | #![warn(clippy::shadow_unrelated)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^
note: initialization happens here
 --> src\agents\tata\mod.rs:6:17
  |
6 |         let x = 2;
  |                 ^
note: previous binding is here
 --> src\agents\tata\mod.rs:4:9
  |
4 |     let x = 1;
  |         ^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#shadow_unrelated
```

</details>

**Clippy version:**

0.0.212 (2020-04-07 d342cee)

**Comment by flip1995** [comment]
This lint has to be disabled at function level. I don't think this can get improved. But we should write it in the documentation of the lint.

**Comment by giraffate** [comment]
> This lint has to be disabled at function level. I don't think this can get improved.

Hi, I have a question. Is this because it's implementing only `check_fn` for `LateLintPass`?
https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/shadow.rs#L99

**Comment by flip1995** [comment]
Yes that is the reason. But I don't know if the lint can be implemented with another method, since we have to look at all nested scopes and the outermost scope (for local vars) in rust is a function.

**Comment by giraffate** [comment]
Ok, thanks for your comment!

**Comment by camsteffen** [comment]
Wonder if this could be fixed after https://github.com/rust-lang/rust/pull/79519 is merged.
